src/libraries/System.Private.CoreLib/src/System/Threading/Lock.cs

```diff
@@ -648,14 +647,22 @@ internal nint LockIdForEvents
 
         internal ulong OwningThreadId => _owningThreadId;
 
-        private static short DetermineMaxSpinCount() =>
-            AppContextConfigHelper.GetInt16Config(
-                "System.Threading.Lock.SpinCount",
-                "DOTNET_Lock_SpinCount",
-                DefaultMaxSpinCount,
-                allowNegative: false);
+        private static short DetermineMaxSpinCount()
+        {
+            if (IsSingleProcessor)
+            {
+                return 0;
+            }
+
+            return
+                AppContextConfigHelper.GetInt16Config(
+                    "System.Threading.Lock.SpinCount",
+                    "DOTNET_Lock_SpinCount",
+                    DefaultMaxSpinCount,
+                    allowNegative: false);
+        }
 
-        private static short DetermineMinSpinCount()
+        private static short DetermineMinSpinCountForAdaptiveSpin()
```

Comment:
If there is a Min and Max spin count - the actual spincount is obviously computed to be somewhere in the range. Not sure the name change makes it more or less clear.

Reply:
The min spin count with adaptive spin is just used as a limit for how low the _spinCount field can go, it can be negative, in which case it's just used as a counter for contentions while skipping spin-waiting until it reaches a value where it tries spin-waiting again. Maybe there's a better name, I can add a comment as well.

Reply:
I think it is ok either way.

`_spinCount` has dual purpose, but the limit is just a limit. 
There are two ways to disable adaptiveness - by setting Min to be equal to Max or by setting it to a negative number. I think it was kind of obvious already.